### PR TITLE
Update the deprecated multicast joinGroup(InetAddress) in the Java 14

### PR DESCRIPTION
**Pull request by valib**

### CODE REVIEW

**Comment by SubJunk** [comment]
When I tested I got the following error, but I need to do more testing to know how that happened

```
15:24:49 INFO  Exception in thread "UPNP-AliveMessageSender" java.lang.NullPointerException
15:24:49 INFO  	at net.pms.network.UPNPHelper.buildMsg(UPNPHelper.java:590)
15:24:49 INFO  	at net.pms.network.UPNPHelper.sendMessage(UPNPHelper.java:375)
15:24:49 INFO  	at net.pms.network.UPNPHelper.sendMessage(UPNPHelper.java:362)
15:24:49 INFO  	at net.pms.network.UPNPHelper.sendAlive(UPNPHelper.java:226)
15:24:49 INFO  	at net.pms.network.UPNPHelper$1.run(UPNPHelper.java:415)
15:24:49 INFO  	at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by valib** [comment]
> 
> 
> When I tested I got the following error, but I need to do more testing to know how that happened
> 
> ```
> 15:24:49 INFO  Exception in thread "UPNP-AliveMessageSender" java.lang.NullPointerException
> 15:24:49 INFO  	at net.pms.network.UPNPHelper.buildMsg(UPNPHelper.java:590)
> 15:24:49 INFO  	at net.pms.network.UPNPHelper.sendMessage(UPNPHelper.java:375)
> 15:24:49 INFO  	at net.pms.network.UPNPHelper.sendMessage(UPNPHelper.java:362)
> 15:24:49 INFO  	at net.pms.network.UPNPHelper.sendAlive(UPNPHelper.java:226)
> 15:24:49 INFO  	at net.pms.network.UPNPHelper$1.run(UPNPHelper.java:415)
> 15:24:49 INFO  	at java.base/java.lang.Thread.run(Unknown Source)
> ```

For me it works without any problemd

File: src/main/java/net/pms/network/UPNPHelper.java [hunk 1]
```diff
@@ -217,12 +217,14 @@ public static void sendAlive() {
 		LOGGER.debug("Sending ALIVE...");
 		MulticastSocket multicastSocket = null;
 		SocketAddress sa = null;
-		NetworkInterface ni = null;
+		NetworkInterface ni = PMS.get().getServer().getNetworkInterface();
+		try {
+			sa = new InetSocketAddress(getIPv4MulticastAddress(), UPNP_PORT);
+		} catch (IOException e1) {
```
Comment (SubJunk) [inline src/main/java/net/pms/network/UPNPHelper.java:223]: Is this an error we want to ignore? What are the consequences of this error if it happens?

Comment (valib) [reply src/main/java/net/pms/network/UPNPHelper.java]: @SubJunk better?

**Comment by valib** [comment]
@SubJunk better?

**Comment by valib** [comment]
@SubJunk now it should be mergeable

**Comment by valib** [comment]
@SubJunk this is worth to be released

**Comment by SubJunk** [comment]
@valib cool I will test it again and merge soon if it works for me

**Comment by SubJunk** [comment]
@valib I have tested this twice now and something is going wrong with hanging/frozen UMS instances. I'm not sure if it is the same in `master` though so I still need to test more. My normal way to use UMS is to force the network interface to be my ethernet card, and for testing this I am setting it to use my wireless card and testing, then setting back to ethernet. I will let you know when I have tested more

**Comment by SubJunk** [comment]
@valib I can't reproduce that problem I had now, so I'm not sure what to do about this. It has left me feeling uneasy about it.
Are there any bugs this change might fix?

**Comment by valib** [comment]
@SubJunk I am not aware about the other bugs this change might fix. For me it works without any problem.

**Comment by valib** [comment]
@SubJunk what about merging this PR?

**Comment by valib** [comment]
@SubJunk is it working for you?

**Comment by SubJunk** [comment]
I will test again. I think the bug I was noticing is already present in master - that using the "restart server" button can cause the application to hang when it is closed - but I will recheck that

**Comment by SubJunk** [comment]
Ok yes I have reproduced the bug on version 9.8.0 

image

File: src/main/java/net/pms/network/UPNPHelper.java [hunk 2]
```diff
@@ -216,11 +216,13 @@ private static void sendReply(String host, int port, String msg) {
...
-
+		SocketAddress sa = null;
+		NetworkInterface ni = null;
 		try {
 			multicastSocket = getNewMulticastSocket();
-			InetAddress upnpAddress = getUPNPAddress();
-			multicastSocket.joinGroup(upnpAddress);
+			sa = new InetSocketAddress(getIPv4MulticastAddress(), UPNP_PORT);
+			ni = PMS.get().getServer().getNetworkInterface();
+			multicastSocket.joinGroup(sa, ni);
```
Comment (SubJunk) [inline src/main/java/net/pms/network/UPNPHelper.java:225]: I know this is already like this in master, but I wonder if we need to keep joining the group every time - can we just stay joined? There seems to be some CPU overhead associated with sending ALIVE messages and I wonder if that is the reason

Comment (valib) [reply src/main/java/net/pms/network/UPNPHelper.java]: @SubJunk this was also my question when I made those changes. Here is the change I made for testing the new approach.

**Comment by valib** [comment]
@SubJunk sorry that I accidentally join this PR with the #2204 but I need it for testing.

**Comment by SubJunk** [comment]
Ok cool, is it ready for me to review/test or are you still working on it?

**Comment by valib** [comment]
@SubJunk testing is welcome.

**Comment by valib** [comment]
@SubJunk now it is ready for testing.

**Comment by SubJunk** [comment]
It seems to work well for me. I joined VLC on one interface, changed the interface and clicked restart, we sent BYEBYE and I couldn't access UMS on VLC anymore, then I changed back to the right interface and clicked UMS restart and I could browse on VLC again.
I also looked in Task Manager and the CPU use seems better, I can see it still goes up to about 0.4% max but that's ok.

Want it merged?

**Comment by valib** [comment]
@SubJunk if you agree it could be merged.

**Comment by valib** [comment]
@SubJunk now is ready to merge.

**Pull request merged**